lib/SILGen/SILGenFunction.h

```diff
@@ -1460,6 +1460,7 @@ class LLVM_LIBRARY_VISIBILITY SILGenFunction
   ManagedValue emitTransformedValue(SILLocation loc, ManagedValue input,
                                     CanType inputType,
                                     CanType outputType,
+                                    bool isFlattening = false,
```

Comment:
I think you should probably just make a new sibling function to create a function-flattening thunk rather than complicate the core value-transformation API like this.